File: Examenes/examen.c
#include <stdio.h> 
#include "funciones.h"

int main(){
    int opcion=0;
    while(1){
    printf("bienvenido al menu las opciones son las siguientes:\n");
    printf("1. verificar si un numero es primo\n");
    printf("2. calcular el factorial de un numero \n");
    printf("3. contar numeros pares e impares\n");
    printf("4. mostrar multiplos de 3 entre 1 y N\n");
    printf("5. salir del programa\n");
    scanf("%d", &opcion);
    switch (opcion){
        case 1:
        int numero;
        printf("Ingrese un número: ");
        scanf("%d", &numero);
        if (EsPrimo(numero)){
        printf("%d no es primo.\n", numero);
        }
        else{   
        printf("%d es primo.\n", numero);}
        

    return 0;
        break;    
        case 2:
            factorial();
            break;
        case 3:
            ContarParesImpares();
            break;
        case 4:
            Multiplosde3();
            break;
        case 5: 
            printf("saliendo del programa....");
            break;
        default:
            printf("ingresa una opcion valida");
    }
    if (opcion == 5){
        printf("Saliendo del programa...\n");
        break;
    }
    }
    return 0;
    }

File: Examenes/tercerParcial/examen.py
from funciones import sumaDivisores, trianguloCaracteres, mostrarPrimerosPrimos, fibonacciInversa
opcion=0
def menu():
    print("""----- MENÚ DE FUNCIONES -----
1. Calcular la suma de todos los divisores de un número N (excluyendo el propio
número).
2. Generar un triángulo de caracteres con letras del alfabeto hasta una altura N.
3. Mostrar los primeros N números primos.
4. Generar la secuencia de los primeros N términos de la serie de Fibonacci
inversa.
5. Salir del programa.
""")
def pedir_datos():
    n= int(input("ingrese un numero: "))
    return n

while True:
    menu()
    opcion= int(input("elija una opcion "))
    if opcion==1:
        n= pedir_datos()
        print(f"la suma de los divisores de {n} es: {sumaDivisores(n)}")
    elif opcion==2:
        n= pedir_datos()
        print("piramide de letras:")
        trianguloCaracteres(n)
    elif opcion==3:
        n= pedir_datos()
        print(f"los primeros {n} numeros primos: ")
        mostrarPrimerosPrimos(n)
    elif opcion==4:
        n= pedir_datos()
        print(f"la serie de fibonacci inversa de {n} terminos: ")
        fibonacciInversa(n)
    elif opcion==5:
        print("saliendo del programa...")
        break

